cmd/csi-provisioner/util.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"hash/fnv"
+)
+
+// copied from https://github.com/openshift/library-go/blob/master/pkg/build/naming/namer.go
```

Comment:
Can you copy the full function description from https://github.com/openshift/library-go/blob/edab248e63516c65a93467eaa8224c86d69f5de9/pkg/build/naming/namer.go#L10-L14?

Reply:
Actually, using https://github.com/openshift/library-go/blob/master/pkg/build/naming/namer.go directly as a dependency may be better. The module doesn't get released (no tags), but it has a compatible license and we only need that one package from it.

Reply:
Yeah, I was afraid I would pull in a large part of Open Shift just for that one function, which I would like to avoid. Let me see what happens.

Reply:
Looks good to me. As expected, changes to go.mod and vendor are minimal. What probably becomes slower are the "go mod" operations when github.com/openshift/library-go hasn't been retrieved yet, but that shouldn't be a concern.

Let's get a second opinion.... @jsafrane: should we fork github.com/openshift/library-go/pkg/build/naming or pull it in as dependency?

Reply:
@jsafrane Could you take a look?

Reply:
openshift/library-go does not guarantee stable API and it does not have tags / releases. While we pull just a small package that we probably won't ever need to re-vendor, I am worried by dumb CVE scanning tools that may catch "unsafe" library-go in go.sum, when the unsafe functionality is actually never used.

I would prefer to copy the function, but I don't have strong opinion.

Reply:
Since we already have both versions it is easy enough to just copy, essentially it will be this https://github.com/kubernetes-csi/external-provisioner/pull/717/commits/995d58b079d6c5958668232e44606f046f2d67b3 but with the format fixes.

Reply:
Let's copy the function.

Reply:
Okay give me a bit to update the PR, I am basically going to revert the last commit, and then fix the small formatting issue

Reply:
As we are back at copying the function, can you address my first comment (copy the function comment, too)?

Reply:
Ahhh of course, sorry about that.